frontend/src/pages/_app.tsx

```diff
@@ -14,37 +14,33 @@ import DatabaseProvider from "../context/DatabaseProvider";
 
 import "bootstrap/dist/css/bootstrap.min.css";
 
-import { LocalizeProvider } from "react-localize-redux";
-import LocalizeWrapper from "../components/utils/LocalizeWrapper";
-import { getI18nConfigForProvider } from "../libraries/language";
+import { getLanguage } from "../libraries/language";
 import { useAnalyticsInit } from "../hooks/analytics/use-analytics-init";
+import { initializeI18n, Language, LANGUAGES } from "../libraries/i18n";
+import { NextComponentType } from "next";
+import { AppContextType, AppPropsType } from "next/dist/shared/lib/utils";
+import { Router } from "next/router";
+import { setI18n } from "react-i18next";
 
 const {
   publicRuntimeConfig: { nearNetworks, googleAnalytics },
 } = getConfig();
 
 type InitialProps = {
-  cookies?: string;
-  acceptedLanguages?: string;
+  language?: Language;
   currentNearNetwork: NearNetwork;
 };
 
-type Props = AppProps & InitialProps;
-
-interface AppType {
-  (props: Props): ReactElement;
-  getInitialProps?: (
-    context: AppContext
-  ) => InitialProps | Promise<InitialProps>;
-}
-
-const App: AppType = ({
-  Component,
-  pageProps,
-  cookies,
-  acceptedLanguages,
-  currentNearNetwork,
-}) => {
+type AppType = NextComponentType<
+  AppContextType<Router>,
+  InitialProps,
+  AppPropsType & InitialProps
+>;
+
+const App: AppType = ({ Component, currentNearNetwork, language }) => {
+  if (typeof window !== "undefined" && language) {
+    initializeI18n(language);
```

Comment:
This code will be called only client-side in case of first render - we'll have language in this case.
In other cases we already have i18n initialized:
- Server-side we initialize it in `getInitialProps`;
- Client-side on subsequent renders we already have everything initialized.